lib/services/flowdock.ts

```diff
@@ -28,6 +28,8 @@ import { ServiceEmitter, ServiceListener } from './service-types';
 
 export class FlowdockService extends Messenger implements ServiceEmitter, ServiceListener, DataHub {
 	private static _serviceName = path.basename(__filename.split('.')[0]);
+	// There are circumstances in which one flowdock event will be received more than once, so track.
+	private postsSynced = new Set<number>();
```

Comment:
Technically, you don't know if you've synced these posts - only that you've added them to a queue to sync hopefully in future. `receivedPostIds`?